src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx

```diff
@@ -0,0 +1,381 @@
+            <StringInputField
+                lineSide={lineSides.left}
+                defaultValue={value}
+                nullValue="[  ]/[   .]"
+                color={value !== undefined ? lineColors.cyan : lineColors.cyan}
+                lsk={LINESELECT_KEYS.L5}
+                selectedCallback={(value) => setNewValue(value)}
+                selectedValidation={((value) => validateEntry(value))}
+                size={lineSizes.regular}
+            />
+        </LineHolder>
+    );
+};
+
+type AdftwoProps = {
+    addMessage: (msg: ScratchpadMessage) => void,
+}
+const Adftwo:React.FC<AdftwoProps> = ({ addMessage }) => {
+    const [value, setValue] = useState<string | undefined>(undefined);
+
+    const setNewValue = (value: string | undefined) => {
+        if (value === undefined) {
+            addMessage(NXSystemMessages.notAllowed);
+        } else {
+            setValue(value);
+        }
+    };
+
+    const validateEntry = (value: string) => {
+        if (value === '') {
+            addMessage(NXSystemMessages.formatError);
+            return false;
+        }
+        return true;
+    };
+    return (
+        <LineHolder columnPosition={2}>
+            <LabelField lineSide={lineSides.right} value="FREQ/ADF2" color={lineColors.white} />
+            <StringInputField
+                lineSide={lineSides.right}
+                defaultValue={value}
+                nullValue="[    .]/[ ]"
+                color={value !== undefined ? lineColors.cyan : lineColors.cyan}
+                lsk={LINESELECT_KEYS.R5}
+                selectedCallback={(value) => setNewValue(value)}
+                selectedValidation={((value) => validateEntry(value))}
+                size={lineSizes.regular}
+            />
+        </LineHolder>
+    );
+};
+
+const Adfonebfo:React.FC = () => (
+    <LineHolder columnPosition={1}>
+        <EmptyLine />
+        <Field
+            lineSide={lineSides.left}
+            value="<-ADF1 BFO"
+            size={lineSizes.regular}
+            color={lineColors.inop}
+        />
+    </LineHolder>
+);
+const Adftwobfo:React.FC = () => (
+    <LineHolder columnPosition={2}>
+        <EmptyLine />
+        <Field
+            lineSide={lineSides.right}
+            value="ADF2 BFO->"
+            size={lineSizes.regular}
+            color={lineColors.inop}
+        />
+    </LineHolder>
+);
+type RadnavePageProps = {
+    addMessage: (msg: ScratchpadMessage) => void,
+}
+
+const RadnavPage: React.FC<RadnavePageProps> = ({ addMessage }) => {
+    const dispatch = useMCDUDispatch();
+    const setTitlebar = ((msg: string) => {
+        dispatch(titlebarActions.setTitleBarText(msg));
+    });
+    useEffect(() => {
+        setTitlebar(' RADIO NAV ');
+    }, []);
+    return (
+        <>
+            <RowHolder columns={2}>
+                <Vorone addMessage={addMessage} />
+                <Vortwo addMessage={addMessage} />
```

Comment:
Consider merging `Vorone` and `vortwo` into one single react component? makes the code tidier and the uses for both cases are the same. The same for `crsone` and `crstwo`, `ls` and `lscrs`, as-well as `adfone` and `adftwo`.

Bear in mind a single component can return `RowHolder`s so with the case of merging `ls` and `lscrs` you can do
```js
return (
    <RowHolder columns={2}>
        <...the inners of ls component/>
     </RowHolder>
     <RowHolder columns={2}>
         <...the innerds of lscrs component />
     </RowHolder>
)
```